modules/virt-adding-public-key-cli.adoc

```diff
@@ -2,13 +2,29 @@
 //
 // * virt/virtual_machines/virt-accessing-vm-ssh.adoc
 
+ifeval::["{context}" == "static-key"]
+:static-key:
+:header: Adding a static key to a VirtualMachine manifest
+endif::[]
+ifeval::["{context}" == "dynamic-key"]
+:dynamic-key:
+:header: Enabling dynamic key injection in a VirtualMachine manifest
+endif::[]
+
 :_content-type: PROCEDURE
-[id="virt-adding-static-public-key-cli_{context}"]
-= Adding an SSH key when creating a virtual machine by using the command line
+[id="virt-adding-public-key-cli_{context}"]
+= {header}
+
+ifdef::static-key[]
+You can add a static public SSH key to a `VirtualMachine` manifest when you create a virtual machine (VM). The key is added to the VM at startup.
 
-You can add a _static_ public SSH key when you create a virtual machine (VM) by using the command line. The key is added to the VM at startup.
+A static key is added to the VM as generated cloud-init metadata, by using a cloud-init configuration disk. This method separates the access credentials from the application data in the cloud-init user data. This method does not affect cloud-init user data.
```

Comment:
@akrejcir is this technically correct?

Reply:
I copied this from upstream docs: https://kubevirt.io/user-guide/virtual_machines/accessing_virtual_machines/#static-ssh-public-key-injection-via-cloud-init

Reply:
Yes it's correct. I've checked the code.